alacritty/src/config/mod.rs

```diff
@@ -318,6 +319,26 @@ pub fn imports(
     Ok(import_paths)
 }
 
+/// Prune the nulls from the YAML.
+fn prune_yaml_nulls(value: &mut serde_yaml::Value) {
+    fn walk(value: &mut serde_yaml::Value) -> bool {
+        if let Some(mapping) = value.as_mapping_mut() {
+            mapping.retain(|_, value| !walk(value));
+            mapping.is_empty()
+        } else if let Some(sequence) = value.as_sequence_mut() {
+            sequence.retain_mut(|value| !walk(value));
+            sequence.is_empty()
+        } else {
+            value.is_null()
+        }
+    }
+
+    if walk(value) {
+        // When the value itself is null return the mapping.
+        *value = serde_yaml::Value::Mapping(Default::default());
+    }
```

Comment:
This seems extremely hacky for effectively not much benefit.

Reply:
That's the only way to translate the empty config though.

Reply:
We could always just not and force users to actually fix their yaml first.

Reply:
Yeah, but empty config is a valid toml config as well, so it seems like our bug.

Reply:
Yeah that's fair, I'm just not sure about this solution in particular. And removing stuff without even printing a log warning definitely shouldn't be happening I think.

Reply:
I can log what I remove though, but null itself is empty, so there's nothing to print, since nothing to remove. it's just a conversion is weird in that case.

Reply:
Yeah with empty config you wouldn't want to log obviously.